pkg/objstore/objstore.proto

```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+
+package objstore;
+
+/* Design questions
+  - Do we want to be able to upload only part of an object
+  - What are we doing for permissions / ACLs
+  - What do we want to do for location, e.g., regions?
+  - How do object versions work?
+  - What concurrency models / consistency guarantees do we support
+  - How are we encoding error codes? Can we use standard rRPC status (https://github.com/grpc/grpc/blob/master/doc/statuscodes.md)
+*/
+
+
+service ObjectStore {
+    rpc createBucket (CreateBucketRequest) returns (Empty) {}
+    rpc listBucket (ListBucketRequest) returns (ListBucketResponse) {}
+    rpc get (GetRequest) returns (GetResponse) {}
+    rpc put (PutRequest) returns (Empty) {}
+    //    rpc deleteBucket
+}
+
+message CreateBucketRequest {
+    string bucketName = 1;
+    // TODO add permissions
+    // TODO add location
+}
+
+message ListBucketRequest {
+    string bucketName = 1;
+}
+
+message ListBucketResponse {
+    repeated string objectName = 1;
+}
+
+message GetRequest {
+    string bucketName = 1;
+    string objectName = 2;
+}
+
+message GetResponse {
+    bytes data = 1;
```

Comment:
Include `objectName` for async request/responses?

Reply:
I don't really understand how gRPC asyc requests work but it seems that there should be some id associated with the request, rather than in the data. There appears to be a provision for tagging the responses, e.g., in [this example](https://github.com/grpc/grpc/blob/v1.23.0/examples/cpp/helloworld/greeter_async_client.cc#L80).

Reply:
You could create separate, async versions of each of the requests, where the request returns a handle that one could wait upon.